csharp/src/Ice/Ice1NetworkSocket.cs

```diff
@@ -323,8 +335,25 @@ internal override ValueTask<SocketStream> SendInitializeFrameAsync(CancellationT
             }
             else
             {
-                return new ValueTask<SocketStream>(CreateStream(false));
+                return new ValueTask<SocketStream>(new Ice1NetworkSocketStream(AllocateId(false), this));
+            }
+        }
+
+        private long AllocateId(bool bidirectional)
+        {
+            // No mutex protection needed, this is called with the mutex locked.
+            long id;
+            if (bidirectional)
+            {
+                id = _nextBidirectionalId;
+                _nextBidirectionalId += 4;
```

Comment:
Needs a comment to explain the +4.

Reply:
We explain some other place in the source that the Ice stream IDs are allocated like Quic stream IDs. I can repeat it again here.